parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -517,7 +524,74 @@ public PageReadStore readNextRowGroup() throws IOException {
     return columnChunkPageReadStore;
   }
 
+  /**
+   * Reads the requested predicate columns dictionaries from the row group at the current file position.
+   * Not reading all the requested columns, only read predicate columns.
+   * eg. select columnA, columnB, columnC from table where columnC > 10
+   *     only reads columnC's dictionary
+   * @param requestedSchema the predicate columns to read
+   * @throws IOException if an error occurs while reading
+   * @return the DictionaryPageReadStore which can provide DictionaryPageReaders for each column.
+   */
+  public DictionaryPageReadStore getCurrentRowGroupDictionaries(MessageType requestedSchema) throws IOException {
```

Comment:
@julienledem @danielcweeks as @danielcweeks explained, the use case is,
read footer, for each row group read blockMetaData and Dictionary for each predicate column chunks, decide whether to read the row group or not based on stats and dictionary, keep the list of to-be-scanned row groups, and then start scanning the necessary row groups.
In this case advance dictionary is separate from advancing row group data.
I am thinking about the following options:
1. advance blockId in both nextRowGroup() and nextRowGroupDictionaries(), while, they should not be called interleaving or together, only one of them could be called in the application
2. have a user level advanceBlock() which will advance blockId, then user could decide when to call nextRowGroup(), when to advanceBlock(), and when to read dictionaries, this could be a little bit too flexible, and existing applications calling nextRowGroup() needs to be updated
3. have a static method in ParquetFileReader, pass BlockMetaData, configuration, path as input parameters
Agree with @danielcweeks option#3 is better. what do you think?